arangod/Replication2/StateMachines/Document/DocumentFollowerState.cpp

```diff
@@ -63,7 +63,10 @@ DocumentFollowerState::DocumentFollowerState(
           handlersFactory->createShardHandler(core->getVocbase(), core->gid)),
       _errorHandler(handlersFactory->createErrorHandler(core->gid)),
       _guardedData(std::move(core), handlersFactory, loggerContext,
-                   _errorHandler) {}
+                   _errorHandler) {
+  // Get ready to replay the log
+  _shardHandler->prepareShardsForLogReplay();
```

Comment:
Does it makes sense to call it here as I see that  this is anyway called after "acquiring" snapshot?

Reply:
Yes, this is actually necessary.

This code place is hit whenever there is a term change (e.g. a LeaderElection / Failover).
As long as followers are in-sync (or close) we will not enter the Snapshot code-path.

The Snapshot code-path is only for new and Empty servers.
Or for servers that runter after a long period of time.

It is also required after the Snapshot.
Because there we first see this constructor.
Then we do the Snapshot.
And without calling this constructor again, we start applying log.
The log may overlap with data from Snapshot.

Reply:
I see. Thx